mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -0,0 +1,192 @@
+from azure.storage.blob import BlobClient
+from azure.core.exceptions import ClientAuthenticationError
+
+import os
+import uuid
+import base64
+import logging
+import requests
+
+from mlflow.entities import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.protos.databricks_artifacts_pb2 import DatabricksMlflowArtifactsService, \
+    GetCredentialsForWrite, \
+    GetCredentialsForRead, ArtifactCredentialType
+from mlflow.protos.service_pb2 import MlflowService, ListArtifacts
+from mlflow.utils.uri import extract_and_normalize_path, is_databricks_acled_artifacts_uri
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.file_utils import relative_path_to_artifact_path, yield_file_in_chunks
+from mlflow.utils.rest_utils import call_endpoint, extract_api_info_for_service
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+
+_logger = logging.getLogger(__name__)
+_PATH_PREFIX = "/api/2.0"
+_AZURE_MAX_BLOCK_CHUNK_SIZE = 100000000  # Max. size of each block allowed is 100 MB in stage_block
+_SERVICE_AND_METHOD_TO_INFO = {
+    service: extract_api_info_for_service(service, _PATH_PREFIX)
+    for service in [MlflowService, DatabricksMlflowArtifactsService]
+}
+
+
+class DatabricksArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on Azure/AWS with access control.
+
+    The artifact_uri is expected to be of the form
+    dbfs:/databricks/mlflow-tracking/<EXP_ID>/<RUN_ID>/artifacts/
+    """
+
+    def __init__(self, artifact_uri):
+        super(DatabricksArtifactRepository, self).__init__(artifact_uri)
+        if not artifact_uri.startswith('dbfs:/'):
+            raise MlflowException('DatabricksArtifactRepository URI must start with dbfs:/')
+        if not is_databricks_acled_artifacts_uri(artifact_uri):
+            raise MlflowException('Artifact URI incorrect. Expected path prefix to be '
+                                  'databricks/mlflow-tracking/path/to/artifact/..')
+        self.run_id = self._extract_run_id()
+
+    def _extract_run_id(self):
```

Comment:
Can I rather make it a helper function in the file, something like https://livegrep.dev.databricks.com/view/mlflow/mlflow/mlflow/store/artifact/dbfs_artifact_repo.py#L147 ?

Reply:
I think a static function within the class makes the most sense, since this is only used within the class itself. The `@staticmethod` decorator may be useful here.

Reply:
Will do!